### ISSUE TITLE
Clarify timing of callbacks for "select" related XRInputSourceEvents

### ISSUE THREAD

**Issue by DsRQuicke** [immersive-web/webxr#950]
*TLDR version*
It might be useful to mention in the [input-explainer.md](https://github.com/immersive-web/webxr/blob/master/input-explainer.md#input-events), or maybe in [hit-testing-explainer.md](https://github.com/immersive-web/hit-test/blob/master/hit-testing-explainer.md#combining-virtual-and-real-world-hit-testing), that any callbacks for the `"selectstart"`, `"select"` and `"selectend"` events are executed right before the rAF callback.

*Long version*
While experimenting (on a mobile phone/tablet) with WebXR and the AR & Hit Test modules, I've had some trouble with the virtual part of ["Combining virtual and real-world hit testing"](https://github.com/immersive-web/hit-test/blob/master/hit-testing-explainer.md#combining-virtual-and-real-world-hit-testing). In my app I use three.js to render my virtual objects, which also includes some small planes that act as buttons. These buttons are kept in sync with the location of the three.js camera, so they appear as a "2D" HUD on the screen.

Initially, I then constructed a ray, using `XRFrame.getPose(evt.inputSource.targetRaySpace)`, in the `"select"` event callback. And used that ray to perform a raycasting on my three.js scene (in that callback). When I touched the buttons, while not moving the device, this worked as expected. But when I moved the device while pushing the buttons, they didn't seem to react. After visualizing a "snapshot" of the ray and the button objects during the `"select"` callback, I noticed that my buttons were lagging behind the ray/touch location (by one frame).

The reason for this one frame delay turned out to be the order in which the `"select"` and the `XRSession.requestAnimation()` callbacks get called. I.e. in a UA "frame" the `"select"` callback gets executed right before the rAF callback. And since my three.js scene and camera are updated in the rAF callback, this means that the camera is still in the old location (from the previous UA "frame" (XRFrame?)) during the `"select"` callback (in the new UA "frame").

![selectFrameDelay](https://user-images.githubusercontent.com/16978270/72385940-cb72c900-3720-11ea-88cc-03b0fb1aa8d6.png)

I solved this by only storing the ray location in the `"select"` callback and postponing the actual raycasting until after the scene & camera get updated in the rAF callback that follows.

I don't know to what extent this is "expected behaviour", or maybe it's specific to the `generic-touchscreen` XRInputSource (or maybe even Chrome (dev v81)). But I thought it might be useful to include some information about the timing of the select/input events in one of the explainers.

**Comment by Manishearth** [comment]
I don't think we want to guarantee anything about the timing here, that's why input events have their own XRFrame attached to them. If you think this is insufficient maybe we can figure out if this can be improved.

**Comment by DsRQuicke** [comment]
Upon further consideration I indeed might have also used the attached xrFrame to update the camera inside the `"select"` callback, something similar to this:
```js
let baseRefSpace = await xrSession.requestReferenceSpace("local");
let viewerRefSpace = await xrSession.requestReferenceSapce("viewer");
let camera = ...;

...

function onSelect(event) {
	let viewerPose = event.frame.getPose(viewerRefSpace, baseRefSpace);

	// Update camera using viewerPose.
	let mat = new THREE.Matrix4().fromArray(viewerPose.transform.matrix);
	mat.decompose(camera.position, camera.quaternion, camera.scale);

	let inputPose = event.frame.getPose(event.inputSource.targetRaySpace, baseRefSpace);

	// Do raycasting against camera HUD using inputPose...
}
```

But in this case I would have some duplicate code, and I would be updating the camera in the `"select"` callback and again in the rAF callback. (And I need to use a `viewerRefSpace`).

On the other hand, once the *DOM overlays* functionality becomes available, the use case I described (i.e. a 2D HUD in 3D space that is fixed to the camera position), will become obsolete in most situations.
So the timing of the events would only be important for very specific use cases, and probably doesn't really warrant an extra clarification in the documentation.
